### ISSUE TITLE
Provide an installer for running in a service account on self hosted Mac

### ISSUE THREAD

**Issue by ericvergnaud** [actions/runner#944]
Thank you 🙇‍♀ for wanting to create a feature in this repository. Before you do, please ensure you are filing the issue in the right place. Issues should only be opened on if the issue **relates to code in this repository**.  

* If you have found a security issue [please submit it here](https://hackerone.com/github)
* If you have questions or issues with the service, writing workflows or actions, then please [visit the GitHub Community Forum's Actions Board](https://github.community/t5/GitHub-Actions/bd-p/actions)
* If you are having an issue or question about GitHub Actions then please [contact customer support](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-github-actions#contacting-support)

If you have a feature request that is relevant to this repository, the runner, then please include the information below:

**Describe the enhancement**
Currently the Github action runner service can only be installed by an interactive user.
This goes against what one expects from a CI i.e. run as a service independently of any user logged in.

**Code Snippet**
Not relevant

**Additional information**
Whilst it may seem a good idea to have 1 user account open a session automatically at startup, and have multiple runners installed under that user account (by running the Github agent runner many times with different names and tokens), it is not practical. Software such as Maven relies on a single .settings.xml tied to the account, so 4 runners running maven in parallel would compete and potentially corrupt the maven repo. This is just one example. We need n accounts in order to have more than 1 runner per self-hosted Mac. And they need to be service accounts in order for the runner to restart automatically.

Creating a service account on a Mac is quite difficult (there is no adduser) but it's possible (see attached).
You can then su into the service account and install the standard runner (see attached).
However, you'll find that the action runner service fails to run.
There are many causes for that: missing dirs (Library/LaunchAgents...), authorizations...

What would be really great would be the following:
 -when initiating creation of a self-hosted runner from the Github UI, propose selection between UI account and service account
- if the latter, propose a downloadable script that can be run using sudo after a chmod 777
- the script itself would receives a few parameters (service group name, user name...)
- it would automate the creation of the service account
- it would automate the creation of the runner, and install a service that:
   - does not rely on interactive account pre-requisites
   - restarts automatically without requiring the user to login
   -

**Comment by ericvergnaud** [comment]
[create-action-runner.txt](https://github.com/actions/runner/files/5879488/create-action-runner.txt)
[create-ci-users.txt](https://github.com/actions/runner/files/5879489/create-ci-users.txt)

**Comment by TingluoHuang** [comment]
@ericvergnaud i deleted your comment as it contains your secret.

**Comment by ericvergnaud** [comment]
Thanks for this.
Shall I open a new issue?

**Comment by TingluoHuang** [comment]
sure.
